src/pages/AdminPages/UnapprovedAccounts/index.js

```diff
@@ -0,0 +1,143 @@
+            setSelectedUser({...selectedUser, role:{
+                isAdmin : false,
+                isSuperAdmin : false,
+                isUser : true,
+                isAffiliate : false,
+                isTrader : false
+            }})
+        }
+        if (e.target.value === 'affiliate') {
+            setSelectedUser({...selectedUser, role:{
+                isAdmin : false,
+                isSuperAdmin : false,
+                isUser : true,
+                isAffiliate : true,
+                isTrader : false
+            }})
+        }
+        if (e.target.value === 'trader') {
+            setSelectedUser({...selectedUser, role:{
+                isAdmin : false,
+                isSuperAdmin : false,
+                isUser : false,
+                isAffiliate : false,
+                isTrader : true
+            }})
+        }
+    }
+
+    const setROI = (e) => {
+        setSelectedUser({...selectedUser, roi: parseFloat(e.target.value)})
+    }
+
+    const setAssignedTrader = (e) => {
+        setSelectedUser({...selectedUser, assignedTrader: e.target.value})
+    }
+
+    useEffect(() => {
+        dispatch(getUnapprovedUsers())
+        // eslint-disable-next-line
+    },[])
+
+    const approveAccount = (user) => {
+        // dispatch(approveUser(selectedUser))
+        console.log(selectedUser)
+    }
+
+    return(
+        <>
+            <Authenticate inside={true} />
+            {
+                !isLoading && role && (
+                role.isAdmin ? (
+                    <>
+                        Unapproved Users
+                        <hr />
+                        {unapprovedUsers.length > 0 && <Table
+                            headers={['firstName', 'lastName', 'email', 'phone', 'approve']}
+                            data={unapprovedUsers.map(user => (
+                                {
+                                    ...user,
+                                    'approve':  <>
+                                        <button onClick={() => {setShowModal(true); setSelectedUser(user)}}>Approve</button>
+                                        <BaseModal 
+                                            title={`Approve ${selectedUser.email}'s account`}
+                                            show={showModal}
+                                            close={() => {setShowModal(false)}}
+                                            doneText="Approve" 
+                                            closeText="Cancel"
+                                            onDone={() => approveAccount(user)}
+                                        >
+                                        <p>Are you sure you want to approve {user.email}'s account?</p>
+                                        <p>First Name: {selectedUser.firstName}</p>
+                                        <p>Last Name: {selectedUser.lastName}</p>
+                                        <p>Email: {selectedUser.email}</p>
```

Comment:
Let's use `Formik` like I did in Login and Register to process all our forms... Thanks

Reply:
Alright